guidelines/project-best-practices.md

```diff
@@ -0,0 +1,197 @@
+  - DCO - [Developer Certificate of Origin](https://github.com/apps/dco)
+  - Unit tests
+  - Integration tests
+  - Scans - see [Security](#Security) section, consider running on schedule (nightly) rather than on each pull request
+  - Note - just because a change passes automated checks doesn't mean it is necessarily good, it still requires judicious maintainer review
+* Test coverage reporting - run on-demand or nightly
+* Keep CI clean and green at all times, address failures and flakes
+* See also proposed [Automated Pipelines task force](https://github.com/hyperledger/toc/issues/44)
+
+## GitHub configuration
+
+* Define repository settings in `.github/settings.yml` so that they can be managed and tracked via pull requests, see [Fabric example](https://github.com/hyperledger/fabric/blob/main/.github/settings.yml).
+* Consider using a CODEOWNERS file to specify write permission per directory
+  - See [Fabric example](https://github.com/hyperledger/fabric/blob/main/CODEOWNERS) with additional `/docs` maintainers.
+  - Add a link from CODEOWNERS to MAINTAINERS.md scope field so that users can find domain area contacts
+* Consider using a .github/[PULL_REQUEST_TEMPLATE.md](https://raw.githubusercontent.com/hyperledger/fabric/main/.github/PULL_REQUEST_TEMPLATE.md) and .github/[ISSUE_TEMPLATE](https://github.com/hyperledger/fabric/tree/main/.github/ISSUE_TEMPLATE)
+* Define Branch protection rules. Recommended configuration:
+  - Check `Require a pull request before merging`
+  - Check `Require approvals`
+  - Check `Dismiss stale pull request approvals when new commits are pushed`
+  - Check `Require review from Code Owners` to enforce CODEOWNERS file.
+  - Check `Require status checks to pass before merging`
+  - Uncheck `Require branches to be up to date before merging` to limit CI runs
+  - Check `Restrict who can push to matching branches`
+  - The remainder can remain unchecked unless there is a specific need
+* Use [Reusable github actions](https://docs.github.com/en/actions/using-workflows/reusing-workflows) to reduce the number of top-level checks
+
+## GitHub workflow
+
+* Although there are often multiple paths to achieve an outcome in git and GitHub, there is value in defining a suggested path for contributors, both for the benefit of new GitHub users, and for the sake of project consistency.
+  - See the [Hyperledger GitHub Contribution Guide](github-contribution-guide.md) for detailed guidance on forking, branching, remotes, creating pull requests, updating pull requests, and cherry picking
+* **Rebase merging** is preferred over **Merge commits** and **Squash merging** to keep commit history and PR description clean (assuming contributors squash/amend their own pull requests)
+* Preserve commit hash - rebase then merge with fast forward option
+  - `git rebase <branch>`; `git merge --ff-only`
+  - `git commit -s` versus `git commit -S`
+    - `-S` is crypto verified, `-s` is DCO signoff. Rebase preserves both.
```

Comment:
@petermetz I think this suggestion came from you, I probably should move it over into the GitHub Contribution Guide at https://github.com/hyperledger/toc/pull/112 , but wasn't sure exactly how to incorporate it. Can you help?

Reply:
@denyeart Sorry for the slow response!
I posted my suggestions on this and also on #112 thank you for considering my suggestions!

Reply:
Added your comments, thanks.